source/MaterialXCore/Types.h

```diff
@@ -12,6 +12,7 @@
 #include <MaterialXCore/Library.h>
 
 #include <array>
+#include <cmath>
```

Comment:
Ideally we should keep references to cmath within Types.cpp, so that this header isn't propagated to all clients.  The performance cost should be very minor, and it keeps our usage of standard math libraries as an implementation detail.

Reply:
This is currently there due to the usage of sqrt in the template. The two options to move it to cpp is to separate the template definition into the cpp file, or add these under the specializations for each type. I think the latter is easier and removes inherited classes list Color3/Color4 from having magnitude and normalization methods which don't really make sense there.